apps/sensenet/src/components/AddButton.tsx

```diff
@@ -173,20 +178,45 @@ export const AddButton: React.FunctionComponent<AddButtonProps> = props => {
       currentComponent || props.path !== '' ? getAllowedChildTypes() : setAllowedChildTypes([])
     }
   }, [
+    currentComponent,
     localization.errorGettingAllowedContentTypes,
     logger,
     parent.Id,
-    currentComponent,
+    personalSettings.uploadHandlers,
     props.path,
     repo,
     showSelectType,
   ])
 
   return (
-    <div className={classes.mainDiv}>
+    <div
+      className={clsx(classes.mainDiv, {
+        [classes.disabled]: !isAvailable,
+      })}>
       {!props.isOpened ? (
         <div className={classes.iconButtonWrapper}>
-          <Tooltip title={localization.addNew} placement="right">
+          {isAvailable ? (
+            <Tooltip
+              title={localization.addNew}
+              placement="right"
+              className={clsx({
+                [classes.disabled]: !isAvailable,
```

Comment:
In `!isAvailable` state it renders another component so this class is not needed IMO.